jupyter_sphinx/execute.py

```diff
@@ -308,11 +351,24 @@ def apply(self):
             kernel_name = nodes[0]['kernel_name'] or default_kernel
             notebook_name = nodes[0]['notebook_name'] or next(default_names)
 
-            notebook = execute_cells(
+            notebook, widget_state = execute_cells(
                 kernel_name,
                 [nbformat.v4.new_code_cell(node.astext()) for node in nodes],
                 self.config.jupyter_execute_kwargs,
             )
+
+            if widget_state:
+                # Append widget state JSON if any widgets were used in the notebook.
+                # XXX: Can we specify a javascript node directly, rather than a 'raw'
+                #      node of 'html' format?
+                doctree.append(docutils.nodes.raw(
+                    text=''.join((
+                        '<script type="application/vnd.jupyter.widget-state+json">',
+                        json.dumps(widget_state),
+                        '</script>')),
+                    format='html'
+                ))
```

Comment:
This means that there will be one `<script>` tag with type `application/vnd.jupyter.widget-state+json` *per notebook that contains widgets* in the page. I assume that the widget IDs will be unique, but I don't know how the ipywidgets javascript handles it when there are several state jsons in the page